src/diffusers/loaders.py

```diff
@@ -946,6 +966,19 @@ def load_lora_weights(self, pretrained_model_name_or_path_or_dict: Union[str, Di
             kwargs (`dict`, *optional*):
                 See [`~loaders.LoraLoaderMixin.lora_state_dict`].
         """
+        # Remove any existing hooks.
+        is_model_cpu_offload = False
+        is_sequential_cpu_offload = False
+        for _, component in self.components.items():
+            if isinstance(component, nn.Module):
+                if hasattr(component, "_hf_hook"):
+                    is_model_cpu_offload = isinstance(getattr(component, "_hf_hook"), CpuOffload)
+                    is_sequential_cpu_offload = isinstance(getattr(component, "_hf_hook"), AlignDevicesHook)
+                    logger.info(
+                        "Accelerate hooks detected. Since you have called `load_lora_weights()`, the previous hooks will be first removed. Then the LoRA parameters will be loaded and the hooks will be applied again."
+                    )
+                    remove_hook_from_module(component)
+
```

Comment:
I think would be ok to move code into a common function called remove_all_accelerate_hooks or something. It could return the two booleans indicating which types of hooks were removed

Reply:
It would be okay. But given it's relatively small in body, I think it's okay to keep it as it helps with readability. Otherwise, it will redirect to a different location. But no strong opinions.

Reply:
Agree with @sayakpaul here, think it's ok as if. Not a fan of a function that has two meanings (a) removing hooks and b) figuring out which hookks existing before => so think it's better to go for readability here

Reply:
ah wasn't aware the snippet was being maintained with copied-from, yeah nw